Lesson_14/Tusk_1.py

```diff
@@ -0,0 +1,115 @@
+import json
+import re
+
+print('Phone list program is started.')
+
+commands = ('You can use next commands:'
+            '\n  - stats: If you want to see the quantity of records'
+            '\n  - add: if you want to add a new record'
+            '\n  - delete <name>: if you want to delete a record with name in <>'
+            '\n  - list: to see the list of all records'
+            '\n  - show <name>: to see the information about the name in <>'
+            '\n  - commands: if you want to see tha command list again'
+            '\n  - exit: if you want to exit the program')
+print(commands)
+file_address = 'phone_book.txt'
+
+try:
+    with open(file_address, 'r') as file:
+        json_data = file.read()
+    phone_list = json.loads(json_data)
+except FileNotFoundError:
+    phone_list = {}
+
+while True:
+    command = input('Please enter a command: ').lower()
+
+    if command == 'stats':
+        if len(phone_list) == 0:
+            print('There is no records for now')
+        else:
+            print(len(phone_list))
+
+    elif command == 'add':
+        name = input('Please enter a name: ')
+        if name in phone_list:
+            print('This name is already here')
+            continue
+
+        while True:
+            phone_nr = input('Please enter a number: ')
+
+            if re.match('\+380[\d]{9}$', phone_nr):
```

Comment:
Also, I would suggest making it simpler and don't overcomplicate:
1. Do not rewrite the original phone number. If user entered `09812345678` let it be the same format 
2. Try to implement DRY principle (Don't Repeat Yourself): here you have 3 conditions with the same code everywhere. So basically, only conditions are different. You can use it and save some space:
```
if re.match(...) or re.match(...) or re.match(...):
    phone_list[name] = phone_nr
    json_data = json.dumps(phone_list)

     with open(file_address, 'w') as file:
         file.write(json_data)
     print(f'The name {name} added')

```

Reply:
yes, it looks better, thank you.